support: update the exterminator working agreement to better document the current process (#10)
windmilleng/company: support/README.md

## Exterminator

Every week, we name an “exterminator” to own user support triage. They triage

> and assign bugs, monitor support channels for new leads, and manage releases.

Reviewer: That sounds like a follow up PR would be good here.

Within a day:

- Email inbox (hi@tilt.dev) (https://groups.google.com/a/tilt.dev/forum/#!forum/hi)

- DMs or Mentions to (@tilt_dev)[https://twitter.com/tilt_dev]

- All new users should be researched and entered into the CRM. If they’re a
  promising lead, the exterminator should reach out to them.

### Support Tracking

> We define "issue" as any user support request that we want to track over

Reviewer: Can we _not_ use "issue" here, since that word is being used to refer to GitHub issues throughout this page.

In particular, we can just call it "user support request". So this section could be:

```
Users requesting support could be because of a bug, a feature request,
a request for docs/examples, or just a general problem with Tilt or using Tilt.

We have two systems for tracking user support requests:

- GitHub issues
- Clubhouse for internal work tracking
```

Author: reworked this a bit!

time. An issue could be a bug, a feature request, a request for docs/examples,
or just a general problem that a user is having trouble solving with Tilt.

We have two systems for tracking issues:

- GitHub issues for all public, user-reported issues

> - Clubhouse for internal task tracking

Reviewer: ```suggestion
- Clubhouse for internal work tracking
```

Let's not use "task". It's a specific feature in Clubhouse that we might want to use more formally in the future.

Author: done

GitHub issues that we are actively working on should be tracked in

> Clubhouse. The reverse is not true - we may have internal work from partners

Reviewer: ```suggestion
Clubhouse. The reverse is not true - we have internal work
```

Let's not say "partners", because it's not defined here, and more broadly, we just internal work for many different reasons.

Author: done

that is not expressed in GitHub issues.

### Support Triage

When a user reports an issue, the Exterminator should ensure there's a Github issue.

The exterminator should confirm with the user that the issue write-up accurately
expresses their problem, and ask them to +1 it or comment with additional context.

We use these votes and comments to decide which issues to prioritize. That
_doesn't_ mean "the issue with the most votes wins". It's just one variable we
use in prioritization.

Issues with a bug label are
automatically added to the [Tilt bug
triage](https://github.com/orgs/windmilleng/projects/2).

Any interesting new users should be mentioned in #triage-customers. The
exterminator should use their best judgment on who is promising enough to DM
directly (based on a combination of research on their background and how active
they are in the channel).

### Internal Support Triage

> When a Tilter notices a bug, they should report it in the #triage-bugs slack

Reviewer: ```suggestion
When a Tilter notices a bug, they should report it in the #triage-bugs Slack
```

- All new users should be researched and entered into the CRM. If they’re a
  promising lead, the exterminator should reach out to them.

### Support Tracking

When users request support, they may be reporting a bug, requesting a feature, asking
for docs/examples, or vaguely describing a general problem that they're having
trouble solving with Tilt.

These user support requests may lead to work that we need to track.

We have two systems for tracking this work:

- GitHub issues for all public, user-reported issues

- Clubhouse for internal tracking and prioritization

GitHub issues that we are actively working on should be tracked in
Clubhouse. The reverse is not true - we may internal work
that is not expressed in GitHub issues.

### Support Triage

When a user reports an issue, the Exterminator should ensure there's a Github issue.

The exterminator should confirm with the user that the issue write-up accurately
expresses their problem, and ask them to +1 it or comment with additional context.

We use these votes and comments to decide which issues to prioritize. That
_doesn't_ mean "the issue with the most votes wins". It's just one variable we
use in prioritization.

> Issues with a bug label are

Reviewer 2: as per [this comment](https://github.com/windmilleng/company/pull/3#issuecomment-592150284)/our new definition of "bug", a. this doc should probably be more deliberate about "support request" instead of bug", and b. is it time to rethink the "bug" vs "enhancement" label?

Reviewer 2: meaning: if it's still baked into our process that the only things we care about fixing asap are "bugs" then we'll either be mislabeling bad UX as bugs in order to fix it, or not fixing bad UX cuz it's not technically a "bug".

(I guess you could make the case that the Exterminator SHOULD only worry about triaging "bugs" [as defined here](https://github.com/windmilleng/company/tree/master/development#bug), but I can think of some things that are poor UX and merit immediate fixing that aren't technically bugs by this definition--e.g. a new feature is causing problems for a specific setup and we need to add a way for users to turn it off, or our logs for something are misleading)

Within a day:

- Email inbox (hi@tilt.dev) (https://groups.google.com/a/tilt.dev/forum/#!forum/hi)

- DMs or Mentions to (@tilt_dev)[https://twitter.com/tilt_dev]

- All new users should be researched and entered into the CRM. If they’re a
  promising lead, the exterminator should reach out to them.

### Support Tracking

When users request support, they may be reporting a bug, requesting a feature, asking
for docs/examples, or vaguely describing a general problem that they're having
trouble solving with Tilt.

These user support requests may lead to work that we need to track.

We have two systems for tracking this work:

- GitHub issues for all public, user-reported issues

- Clubhouse for internal tracking and prioritization

GitHub issues that we are actively working on should be tracked in

> Clubhouse. The reverse is not true - we may internal work

Reviewer: ```suggestion
Clubhouse. The reverse is not true - we have internal work
```

the [Tilt bug triage](https://github.com/orgs/windmilleng/projects/2).

Any interesting new users should be mentioned in #triage-customers. The
exterminator should use their best judgment on who is promising enough to DM
directly (based on a combination of research on their background and how active
they are in the channel).

### Internal Support Triage

When a Tilter has a problem with Tilt, they should report it in the #triage-bugs Slack
channel.

The exterminator should determine whether the problem belongs in Github Issues or
Clubhouse, based on whether users have noticed and who it affects, and file it
in the appropriate place.

## Request Priority

The exterminator picks which support requests are high-priority to deserve
immediate attention. Factors include:

- Is there a workaround? (If so, the exterminator owns verifying the user is able to use it)

- How bad is the impact? Are there side effects (like data loss in the cluster)

- How common is it?

- Is it blocking a high-priority team?

If you aren't sure, ask a teammate or in #triage-bugs.

For high priority requests, the exterminator should assign someone to start work
immediately.

> Low priority requests should be tracked in clubhouse and teams should make their own

Reviewer: ```suggestion
Low priority requests should be tracked in Clubhouse and teams should make their own
```

judgements about how to prioritize them. If the exterminator sees a low priority

> request that they know how to fix quickly, they have the authority to take action.

Reviewer: ```suggestion
request that they know how to fix quickly, they should consider fixing it right away.
```

I don’t like the word “authority”. If mis-understood, one may think Tilt engineers don’t have authority to take action under general circumstances. I’d like to think Tilters should have freedom to pick what to work on, guide by documents like this to make good judgement.

Author: done!

- Is it blocking a high-priority customer?

If you aren't sure, ask a teammate or in #triage-bugs.

For high priority bugs, the exterminator should assign someone to start work immediately.

> Low priority bugs should be tracked in clubhouse and teams should make their own

Reviewer: ```suggestion
Low priority bugs should be tracked in Clubhouse and teams should make their own
```
